### PR TITLE
Reinit of structure sensor_data while using it causing functionality break on versal

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
structure sensor_data is already initialized with defaults during its declaration. Also, data_type has default constructor that initializes data members appropriately, so use "= {0}" for fundamental types only.
Due to assigning of 0 to data_type, next line is not getting executed and hence xbutil reports data is not displaying properly on vck5000 platform. This issue is introduced with PR #6602 .
Signed-off-by: Rajkumar Rampelli <rajkumar@xilinx.com>

#### Problem solved by the commit
Seeing no proper electrical output in xbutil reports on vck5000. Tried "xbutil examine -r electrical -d" -> shows no electrical info.
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
This issue identified while validating vck5000 features using latest XRT package. It is a regression due to PR #6602.
#### How problem was solved, alternative solutions (if any) and why they were rejected
Added changes to fix this issue. Reinit of structure sensor_data causing this issue. This code was introduced as part of PR #6602

#### Risks (if any) associated the changes in the commit
NA
#### What has been tested and how, request additional testing if necessary
xbutil examine reports.
#### Documentation impact (if any)
NA

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
**With this PR:**
`xbutil examine -r electrical -d

-------------------------------------------------------
1/1 [0000:d8:00.1] : xilinx_vck5000_gen4x8_xdma_base_2
-------------------------------------------------------
Electrical
  Max Power              : NA Watts
  Power                  : 27 Watts
  Power Warning          : NA

  Power Rails            : Voltage   Current
  12v_pex                : 12.032 V,  0.887 A
  3v3_pex                :  3.301 V
  3v3_aux                :  3.317 V
  12v_aux_0              : 11.981 V,  1.012 A
  12v_aux_1              : 11.960 V,  0.553 A
  vccint                 :  0.805 V, 148.800 A
                         :  0.000 V,  0.000 A
`

**Using main XRT:**
`xbutil examine -r electrical -d

-------------------------------------------------------
1/1 [0000:d8:00.1] : xilinx_vck5000_gen4x8_xdma_base_2
-------------------------------------------------------
Electrical
  Max Power              : NA Watts
  Power                  : NA Watts
  Power Warning          : NA

  Power Rails            : Voltage   Current
`

File: src/runtime_src/core/pcie/linux/device_linux.cpp [hunk 1]
```diff
@@ -149,7 +149,7 @@ struct sdm_sensor_info
     std::array<std::string, 6> sname_end = {"label", "input", "max", "average", "highest", "status"};
     int max_end_types = sname_end.size();
     std::string errmsg, str_op, target_snode;
-    data_type data = { 0 };
+    data_type data;
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/pcie/linux/device_linux.cpp:152]: is this happening due to initializing string members also to zero? I didn't quite understand the fix.

Comment (stsoe) [reply src/runtime_src/core/pcie/linux/device_linux.cpp:152]: I agree with this fix,  data_type has default constructor that initializes data members appropriately,. `= {0}` only makes sense for fundamental types.  The PR comment doesn't exactly match but I get it, thanks!

**Review by stsoe — APPROVED** [review]

File: src/runtime_src/core/pcie/linux/device_linux.cpp (hunk 1 above)
Comment (keryell) [reply src/runtime_src/core/pcie/linux/device_linux.cpp:152]: This deserves a comment.

Comment (chvamshi-xilinx) [reply src/runtime_src/core/pcie/linux/device_linux.cpp:152]: = {0} doesn't make sense for this structure type, but this structure is getting filled up below this code. I just wanted to understand is there any other potential bug in the code.

Comment (stsoe) [reply src/runtime_src/core/pcie/linux/device_linux.cpp:152]: I think the proper fix is 
`data_type data {};`

**Comment by dayeh-xilinx** [comment]
retest this please. Failed to retrieve files from:https://github.com/Xilinx/ERT-BSP/raw/main/BSPs/sched_bsp.tar.bz2

File: src/runtime_src/core/pcie/linux/device_linux.cpp (hunk 1 above)
Comment (rajkumar-xilinx) [reply src/runtime_src/core/pcie/linux/device_linux.cpp]: > I think the proper fix is `data_type data {};`

Yes, this change is worked.

**Review by chvamshi-xilinx — APPROVED** [review]

**Pull request merged**